flink-core/src/main/java/org/apache/flink/api/common/operators/ResourceSpec.java

```diff
@@ -198,26 +218,34 @@ public double getGPUResource() {
 	}
 
 	public Map<String, Resource> getExtendedResources() {
+		throwUnsupportedOperationExecptionIfUnknown();
 		return extendedResources;
 	}
 
+	private void throwUnsupportedOperationExecptionIfUnknown() {
+		if (this == UNKNOWN) {
+			throw new UnsupportedOperationException();
+		}
+	}
+
 	/**
 	 * Check whether all the field values are valid.
 	 *
 	 * @return True if all the values are equal or greater than 0, otherwise false.
 	 */
 	public boolean isValid() {
-		if (this.cpuCores >= 0 && this.heapMemoryInMB >= 0 && this.directMemoryInMB >= 0 &&
-				this.nativeMemoryInMB >= 0 && this.stateSizeInMB >= 0 && managedMemoryInMB >= 0) {
-			for (Resource resource : extendedResources.values()) {
-				if (resource.getValue() < 0) {
-					return false;
-				}
-			}
+		if (this == UNKNOWN) {
```

Comment:
I think we should use `equals` instead of `==`.
`UNKNOWN` resource spec can be initialized in different nodes and created by deserialization, which means there can be multiple `UNKNOWN` instances.

Reply:
Actually, `UNKNOWN` is guaranteed singleton. See `ResourceSped#readResolve`.

Reply:
That's right. But I thin using it like this is causing confusions and should be avoided if possible. Another case is "== ANY" in ResourceProfile(while ANY is not handled in ResourceProfile#readResolve), see my other comment.

Reply:
Nice catch. We should also handle `ANY` in `ResourceProfile#readResolve`, to make sure it's singleton.
And yes, it makes sense to use `equals` rather than `==`, for clarity and for eliminating dependencies on the singleton property. I'll make the change.